heat/sparse/dcsr_matrix.py

```diff
@@ -0,0 +1,332 @@
+"""Provides Dcsr_matrix, a distributed compressed sparse row matrix"""
+from __future__ import annotations
+
+import torch
+
+from mpi4py import MPI
+from typing import Union, Tuple, TypeVar
+
+__all__ = ["Dcsr_matrix"]
+
+Communication = TypeVar("Communication")
+
+
+class Dcsr_matrix:
```

Comment:
Distributed CSR Matrix class.

Stores the local array in the form of `torch.sparse_csr_tensor`